### PR TITLE
RList: added remove_last, remove_eq, update_eq and for_all functions.

### PR DESCRIPTION
**Pull request by sagotch**

### CODE REVIEW

File: src/reactiveData.ml [hunk 1]
```diff
@@ -454,6 +468,21 @@ module RList = struct
...
+    let (res, set_res) = React.S.create (List.for_all fn (value data)) in
+    ignore @@ React.E.map (function
+        | Set x ->
+          set_res (List.for_all fn x) ;
+        | Patch updates ->
+          List.iter
+            (function
+              | R _ | X _ -> ()
+              | I (_, v) | U (_, v) ->
+                set_res ((React.S.value res) && fn v) )
```
[the remark this answers is not in the document]
Comment (sagotch) [reply src/reactiveData.ml]: Also, `R _ | X _ -> ()` is wrong...

File: src/reactiveData.ml [hunk 2]
```diff
@@ -454,6 +473,22 @@ module RList = struct
     in
     from_event (List.rev (value t)) e
 
+  let for_all fn data =
```
Comment (sagotch) [inline src/reactiveData.ml]: Does this version seem OK @vasilisp?

File: src/reactiveData.ml [hunk 3]
```diff
@@ -454,6 +473,22 @@ module RList = struct
...
+    let f = fun acc -> function
+      | Set x ->
+        List.for_all fn x
+      | Patch updates ->
+        let rec iter acc = function
+          | [] -> acc
+          | X _ :: tl -> iter acc tl
+          | R _ :: tl -> iter (if acc then acc else List.for_all fn (value data)) tl
```
Comment (sagotch) [inline src/reactiveData.ml]: I did not find anything better than `List.for_all fn (value data))` when removing a element which was not satisfying `fn`.

File: src/reactiveData.ml [hunk 4]
```diff
@@ -473,25 +473,45 @@ module RList = struct
...
-               if acc then acc && fn v else List.for_all fn (value data) )
+             | X (i, i') ->
+               if IntSet.mem i acc = IntSet.mem i' acc
+               then acc
+               else if IntSet.mem i acc
+               then IntSet.add i' (IntSet.remove i acc)
+               else IntSet.add i (IntSet.remove i' acc)
+             | R i ->
+               let acc = update_idx_after i (-) acc in
+               IntSet.remove i acc
```
Comment (vouillon) [inline src/reactiveData.ml]: I think you need to call `IntSet.remove` before calling `update_idx_after`

Comment (sagotch) [reply src/reactiveData.ml]: Indeed.

**Pull request merged**